scheduler/src/test/java/org/cloudfoundry/autoscaler/scheduler/rest/ScheduleRestController_SpecificSchedule_Test.java

```diff
@@ -405,23 +352,50 @@ public void testCreateScheduleWithOverlappingSpecificDateSchedule_Failure_16() t
 		assertUserError(resultActions, status().isBadRequest(), errorMessage);
 	}
 
-	private ResultActions getResultActionWithCreateSchedule(int noOfSpecificDateSchedulesToSetUp,
-			int noOfRecurringSchedulesToSetUp) throws Exception {
+	private String getCreateSchedulePath() {
+		return String.format("/v2/schedules/%s", appId);
+	}
+
+	private ResultActions callCreateSchedules(int noOfSpecificDateSchedulesToSetUp, int noOfRecurringSchedulesToSetUp)
+			throws Exception {
 		String content = TestDataSetupHelper.generateJsonSchedule(noOfSpecificDateSchedulesToSetUp,
 				noOfRecurringSchedulesToSetUp);
 
 		return mockMvc.perform(put(getCreateSchedulePath()).contentType(MediaType.APPLICATION_JSON).content(content));
 
 	}
 
-	private ResultActions getResultActionWithGetSchedule() throws Exception {
+	private ResultActions callGetAllSchedulesByAppId() throws Exception {
 
 		return mockMvc.perform(get(getCreateSchedulePath()).accept(MediaType.APPLICATION_JSON));
 
 	}
 
-	private String getCreateSchedulePath() {
-		return String.format("/v2/schedules/%s", appId);
+	private void assertCreateAndGetSchedules(int expectedSchedulesTobeFound) throws Exception {
+
+		ResultActions resultActions = callCreateSchedules(expectedSchedulesTobeFound, 0);
+		assertCreateScheduleAPI(resultActions);
+
+		resultActions = callGetAllSchedulesByAppId();
+		assertSpecificSchedulesFoundEquals(expectedSchedulesTobeFound, resultActions);
+
+		// reset all records for next test.
+		afterTest();
```

Comment:
There should be no direct call to afterTest()
Your cleanup is done by the test framework after the test is truly done.
If this is needed, then the tests are wrong and you have combined multiple tests together.